api/src/main/java/org/apache/iceberg/actions/RewriteTablePath.java

```diff
@@ -112,5 +122,15 @@ interface Result {
 
     /** Name of latest metadata file version */
     String latestVersion();
+
+    /** count of rewrite delete files, default value is 0 */
+    default int rewrittenDeleteFilesCount() {
```

Comment:
I think rewrittenDeleteFilesCount might be a bit misleading? Aren't we just rewriting table paths and not actually compacting files together? Other Actions/Procedures use a similar naming but there it actually means that files have been compacted together